bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,483 @@
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("batteryChargeTimeTo")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556240", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, batteryChargeTimeTo !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("maxDepthOfDischarge")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556247", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, maxDepthOfDischarge !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("shadowManagement")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556483", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug("Unexpected error in KostalConfigurationHandler.configurationHandler, shadowManagement !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("externalModuleControl")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556482", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, externalModuleControl !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("inverterName")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "16777984", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug("Unexpected error in KostalConfigurationHandler.configurationHandler, inverterName !", e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SecondGenerationConfiguration.class);
+
+        scheduler.scheduleWithFixedDelay(() -> {
```

Comment:
You should call this at the end of your `initialize` method otherwise you run the risk of `refresh` being called before you have populated your channel configs.